sdk/tests/extra/workload-simulator.html

```diff
@@ -0,0 +1,604 @@
+    gl.shaderSource(vs, document.getElementById('vs').textContent);
+    gl.compileShader(vs);
+    //console.log(gl.getShaderInfoLog(vs));
+    program = gl.createProgram();
+    gl.attachShader(program, fs);
+    gl.attachShader(program, vs);
+    gl.linkProgram(program);
+    //console.log(gl.getProgramInfoLog(program));
+    gl.useProgram(program);
+    program.position = gl.getAttribLocation(program, 'position');
+    program.texCoordIn = gl.getAttribLocation(program, 'texCoordIn');
+    program.offset = gl.getUniformLocation(program, 'offset');
+    program.tex = gl.getUniformLocation(program, 'tex');
+
+    // Setup vertex buffer
+    buffer = gl.createBuffer();
+    gl.bindBuffer(gl.ARRAY_BUFFER, buffer);
+    gl.bufferData(gl.ARRAY_BUFFER, new Float32Array(vertices), gl.STREAM_DRAW);
+    gl.enableVertexAttribArray(program.position);
+    gl.vertexAttribPointer(program.position, 2, gl.FLOAT, false, 0, 0);
+
+    // Setup drawing state
+    gl.viewport(0, 0, webgl.width, webgl.height);
+    gl.uniform1i(program.tex, 0);
+    gl.clearColor(1, 1, 1, 1);
+    gl.disable(gl.SCISSOR_TEST);
+    gl.disable(gl.DEPTH_TEST);
+    gl.disable(gl.STENCIL_TEST);
+    gl.disable(gl.BLEND);
+    gl.disable(gl.CULL_FACE);
+  }
+
+  gl.clear(gl.COLOR_BUFFER_BIT);
+
+  // Upload some data to test the PCI bottleneck.
+  if (uploads.value > 0) {
+    if (writeBufferArray.length * 4 != uploads.value * 1024 * 1024) {
+      writeBufferArray = new Float32Array(uploads.value * 1024 * 1024 / 4);
+      // We want to actually use this data in rendering so the graphics driver
+      // can't optimize away the upload. Fill the first few bytes with our real
+      // vertex data.
+      writeBufferArray.set(vertices, 0);
+    }
+    gl.bufferData(gl.ARRAY_BUFFER, writeBufferArray, gl.STREAM_DRAW);
+  }
+
+  // Actually draw the map.
+  gl.uniform2f(program.offset,
+               (position[0] - imgSize) / webgl.width * 2,
+               -position[1] / webgl.width * 2);
+  const numDrawCalls = Math.max(1, drawCalls.value * multiplier);
+  const instances = pixels.value / 64000 * multiplier;
+  const instancesPerCall = Math.max(1, instances / numDrawCalls | 0);
+  const instancesFirstCall = instancesPerCall + numDrawCalls % instancesPerCall;
+  gl.drawArraysInstanced(gl.TRIANGLE_STRIP, 0, 4, instancesFirstCall);
+  var instancesDrawn = instancesFirstCall;
+  // Set up to scissor out all but one pixel of the map.
+  gl.scissor(position[0], webgl.height - position[1] - 1, 1, 1);
+  let scissorEnabled = false;
+  for (let i = 1; i < numDrawCalls; i++) {
+    if (instancesDrawn > instances && !scissorEnabled) {
+      // If we've drawn all of the requested pixels already, enable the scissor
+      // test so we only draw one pixel per draw call for the rest of the calls.
+      scissorEnabled = true;
+      gl.enable(gl.SCISSOR_TEST);
+    }
+    gl.drawArraysInstanced(gl.TRIANGLE_STRIP, 0, 4, instancesPerCall);
+    instancesDrawn += instancesPerCall;
```

Comment:
Have you observed on some platforms that the additional pixels drawn are having an effect? On my MacBook Pro they seem to be being short-circuited by the driver or something similar, and they don't cause a slowdown.

Reply:
I think the machine is just too fast. Try "Multiply the above slider values by: 10"

I put the default maximum values pretty low for testing on low-end Android phones. Also because on some platforms it's pretty easy to hose your machine, as Gregg just discovered :neutral_face: